src/oscar/apps/offer/abstract_models.py

```diff
@@ -940,7 +940,7 @@ def product_queryset(self):
 
             # select all those product that are selected either by product class,
             # category, or explicitly by included_products.
-            selected_parents = Product.objects.annotate(
```

Comment:
Renaming this variable because `selected_parents` is misleading - this queryset includes all structures.